src/mode/behaviour/cstyle.js

```diff
@@ -357,53 +360,52 @@ CstyleBehaviour.isSaneInsertion = function(editor, session) {
     return iterator.getCurrentTokenRow() !== cursor.row ||
         this.$matchTokenType(iterator.getCurrentToken() || "text", SAFE_INSERT_BEFORE_TOKENS);
 };
-
-CstyleBehaviour.$matchTokenType = function(token, types) {
+CstyleBehaviour["$matchTokenType"] = function(token, types) {
     return types.indexOf(token.type || token) > -1;
 };
 
-CstyleBehaviour.recordAutoInsert = function(editor, session, bracket) {
+CstyleBehaviour["recordAutoInsert"] = function(editor, session, bracket) {
```

Comment:
I would rather aim to make it work as it is, without trying to please TS language server too much by rewriting everything into ["..."]. You probably just need to define a proper typedef.

Reply:
Generally I agree with you, but lets leave it for further improvements (especially whole `mode/*` part)